tests/unittests/import_export_tests.py

```diff
@@ -0,0 +1,1675 @@
+            dataset2.distinct("predictions.detections.confidence"),
+        )
+        self.assertEqual(
+            dataset.distinct("predictions.detections.age"),
+            dataset2.distinct("predictions.detections.age"),
+        )
+        self.assertEqual(
+            dataset.distinct("predictions.detections.cute"),
+            dataset2.distinct("predictions.detections.cute"),
+        )
+        self.assertEqual(
+            dataset.distinct("predictions.detections.mood"),
+            dataset2.distinct("predictions.detections.mood"),
+        )
+
+        # Omit extra attributes
+
+        export_dir = self._new_dir()
+
+        dataset.export(
+            export_dir=export_dir,
+            dataset_type=fo.types.COCODetectionDataset,
+            extra_attrs=False,
+        )
+
+        dataset2 = fo.Dataset.from_dir(
+            dataset_dir=export_dir,
+            dataset_type=fo.types.COCODetectionDataset,
+            label_field="predictions",
+        )
+
+        self.assertEqual(dataset2.distinct("predictions.detections.age"), [])
+        self.assertEqual(dataset2.distinct("predictions.detections.cute"), [])
+        self.assertEqual(dataset2.distinct("predictions.detections.mood"), [])
+
+        # Labels-only
+
+        data_path = self.images_dir
+        labels_path = os.path.join(self._new_dir(), "labels.json")
+
+        dataset.export(
+            dataset_type=fo.types.COCODetectionDataset,
+            labels_path=labels_path,
+        )
+
+        dataset2 = fo.Dataset.from_dir(
+            dataset_type=fo.types.COCODetectionDataset,
+            data_path=data_path,
+            labels_path=labels_path,
+            label_field="predictions",
+        )
+
+        self.assertEqual(len(dataset), len(dataset2))
+        self.assertSetEqual(
+            set(dataset.values("filepath")), set(dataset2.values("filepath")),
+        )
+        self.assertEqual(
+            dataset.count("predictions.detections"),
+            dataset2.count("predictions.detections"),
+        )
+
+    @drop_datasets
+    def test_voc_detection_dataset(self):
+        dataset = self._make_dataset()
+
+        # Standard format
+
+        export_dir = self._new_dir()
+
+        view = dataset.limit(2)
+        view.export(
+            export_dir=export_dir, dataset_type=fo.types.VOCDetectionDataset,
+        )
+
+        dataset2 = fo.Dataset.from_dir(
+            dataset_dir=export_dir,
+            dataset_type=fo.types.VOCDetectionDataset,
+            label_field="predictions",
+        )
+
+        self.assertEqual(len(view), len(dataset2))
+        self.assertEqual(
+            view.count("predictions.detections"),
+            dataset2.count("predictions.detections"),
+        )
+        self.assertEqual(
+            view.distinct("predictions.detections.age"),
+            dataset2.distinct("predictions.detections.age"),
+        )
+        self.assertEqual(
```

Comment:
This is failing, as booleans are converted strings when outputting as XML.

```
======================================================================
FAIL: test_voc_detection_dataset (__main__.ImageDetectionDatasetTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ben/code/fiftyone/tests/unittests/decorators.py", line 21, in wrapper
    return func(*args, **kwargs)
  File "import_export_tests.py", line 686, in test_voc_detection_dataset
    self.assertEqual(
AssertionError: Lists differ: [False, True] != ['False', 'True']

First differing element 0:
False
'False'

- [False, True]
+ ['False', 'True']
?  +     +  +    +
```

Reply:
Since there's no way to tell if the booleans in an XML are meant to be booleans or just strings, do you think it would be better to convert them to `bool` automatically or just leave them as strings?

Reply:
I'd say that importing strings is fine for formats that don't have an option for boolean data types

Reply:
Yes, I'm fine with the current behavior. I don't know VOC, but if they don't support booleans I think this is a good best effort. XML itself is certainly capable of holding booleans, if we wanted to look into that at some point.

So a user will have to convert booleans after import. FWM

Reply:
Oops I tried to implement this such that things would be converted to more specific types than string, if possible, during import. But I guess I messed that up for VOC? I think it is working for other XML types like CVAT. Take a look there for copy-paste code?

Reply:
> Oops I tried to implement this such that things would be converted to more specific types than string, if possible, during import. But I guess I messed that up for VOC? I think it is working for other XML types like CVAT. Take a look there for copy-paste code?

Ok, cool. I'll try to adjust.

Reply:
I see the same issue for CVAT.

```py
import fiftyone as fo

dataset = fo.Dataset()
sample = fo.Sample("/home/ben/data/num-1-100/001.png")
sample["detection"] = fo.Detection(boolean=True, bounding_box=[0, 0, 1, 1]) 
dataset.add_sample(sample)

dataset.export(
    export_dir="~/tmp/cvat-test-export", dataset_type=fo.types.CVATImageDataset
)

dataset2 = fo.Dataset.from_dir(
    dataset_dir="/tmp/cvat-test-export",
    dataset_type=fo.types.CVATImageDataset,
    label_field="predictions",
)

print(dataset2.first())
```

Out:
```
<Sample: {
    'id': '60ff2a4fb11c60c518138e17',
    'media_type': 'image',
    'filepath': '/tmp/cvat-test-export/data/001.png',
    'tags': BaseList([]),
    'metadata': <ImageMetadata: {
        'size_bytes': None,
        'mime_type': None,
        'width': 128,
        'height': 128,
        'num_channels': None,
    }>,
    'predictions_detections': <Detections: {
        'detections': BaseList([
            <Detection: {
                'id': '60ff2a4fb11c60c518138e16',
                'attributes': BaseDict({}),
                'tags': BaseList([]),
                'label': 'None',
                'bounding_box': BaseList([0.0, 0.0, 1.0, 1.0]),
                'mask': None,
                'confidence': None,
                'index': None,
                'boolean': 'True',
            }>,
        ]),
    }>,
}>
```